File: D-ETCIF-backend/internal/model/help.go
// Package model
// D-ETCIF-backend/internal/model/help.go
package model

type HelpDetail struct {
	ID              int    `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID          int    `gorm:"not null" json:"user_id"`
	ExperimentID    int    `gorm:"not null" json:"experiment_id"`
	ExperimentStage int    `gorm:"not null" json:"experiment_stage"`
	Title           string `gorm:"type:varchar(50)" json:"title"`
	Content         string `gorm:"type:varchar(255)" json:"content"`
	Status		  int    `gorm:"not null" json:"status"` // 0: 待处理, 1: 已处理
	CreatedAt       int64  `gorm:"autoCreateTime" json:"created_at"`
}

File: D-ETCIF-frontend/src/types/help.ts
export interface HelpDetail {
  id: number;
  user_id: number;
  username?: string; // 可选，如果后端做了关联查询建议带上
  experiment_id: number;
  experiment_stage: number | string; // 兼容后端数值阶段与历史字符串阶段
  title: string;
  content: string;
  status?: number; // 0: 待处理, 1: 已解决 (建议预留)
  created_at?: number; // Unix 时间戳
}
